pkg/fqdn/cache.go

```diff
@@ -154,9 +162,56 @@ func (c *DNSCache) updateWithEntry(entry *cacheEntry) {
 		c.forward[entry.Name] = entries
 	}
 	c.updateWithEntryIPs(entries, entry)
-	// When lookupTime is much earlier than time.Now(), we may not expire all
-	// entries that should be expired, leaving more work for .Lookup.
-	c.removeExpired(entries, time.Now())
+
+}
+
+// AddNameToCleanup adds the IP with the given TTL to the the cleanup map to
+// delete the entry from the policy when it expires.
+// Need to be called with the lock
+func (c *DNSCache) addNameToCleanup(entry *cacheEntry) {
```

Comment:
We should avoid adding this to cleanup if `entry.ExpirationTime` is before `c.lastCleanup`

Reply:
How can that be possible? the TTL is time.Now() + TTL no?